plugin/download.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+# download.sh <tag-name>
+
+set -o errexit    # always exit on error
+set -o pipefail   # don't ignore exit codes when piping output
+
+TAG_NAME=${1:-}
+PLUGIN_DIR=$(dirname ${0:-})
```

Comment:
That's redundant; `${0:-}` would expand to nothing, and `$0` can be assumed to be there (if it's not, you'd want this to fail most likely, because then something is really wrong with the shell).